File: Swift/Stanford/ObjectiveC/Matchismo/Matchismo/Matchismo/Model/Card.m
//
//  Card.m
//  Matchismo
//
//  Created by Tom Lai on 24/3/2018.
//  Copyright © 2018年 Tom Lai. All rights reserved.
//

#import "Card.h"

@interface Card()

@end

@implementation Card

- (int) match:(NSArray*) otherCards{
    
    int score = 0;
    for(Card *card in otherCards){
        if([card.contents isEqualToString:self.contents]){
            score = 1;
        }
    }
    
    return score;
}

@end

File: Swift/Stanford/Swift/Set/Set/Model/Card.swift
//
//  Card.swift
//  Set
//
//  Created by Tom Lai on 31/3/2018.
//  Copyright © 2018年 Tom Lai. All rights reserved.
//

import Foundation

class Card{
    
    var isFaceUp = false
    var isMatched = false
    var identifier: Int
    
    static var identifierFactory = 0
    
    static func getUnqiueIdentifier() -> Int{
        identifierFactory += 1
        return identifierFactory
    }
    
    init(){
        self.identifier = Card.getUnqiueIdentifier();
    }
}

